src/JWT/Builder/JwtBuilder.cs

```diff
@@ -247,6 +249,22 @@ public string Encode()
             return _encoder.Encode(_jwt.Header, _jwt.Payload, _secrets?[0]);
         }
 
+        public string Encode<T>(T payload)
+        {
+            EnsureCanEncode();
+
+            var dic = payload.GetType()
+                             .GetProperties(BindingFlags.Instance | BindingFlags.Public)
+                             .ToDictionary(prop => prop.Name, prop => prop.GetValue(payload, null));
+
+            foreach (var pair in dic)
+            {
+                _jwt.Payload.Add(pair.Key, pair.Value);

```

Comment:
Should we add method `AddRange()` to avoid this loop?

If not, please see if you can deconstruct to a value tuple `(key, value)`

Reply:
I'll look into it

Reply:
Payload is a dictionary so it has no AddRange
https://stackoverflow.com/questions/6050633/why-doesnt-dictionary-have-addrange